### PR TITLE
Use Sources RuntimeEnvironment

### PR DESCRIPTION
**Pull request by BrennanConroy**
@pranavkm @JunTaoLuo

### CODE REVIEW

File: src/Microsoft.AspNetCore.Hosting/Startup/StartupExceptionPage.cs [hunk 1]
```diff
@@ -243,13 +242,28 @@ private static string GenerateFooterEncoded()
...
-            var runtimeArch = HtmlEncodeAndReplaceLineBreaks(environment.RuntimeArchitecture);
+            var runtimeArch = HtmlEncodeAndReplaceLineBreaks(RuntimeInformation.ProcessArchitecture.ToString());
             var currentAssembly = typeof(StartupExceptionPage).GetTypeInfo().Assembly;
             var currentAssemblyVersion = currentAssembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>().InformationalVersion;
             currentAssemblyVersion = HtmlEncodeAndReplaceLineBreaks(currentAssemblyVersion);
 
-            var os = HtmlEncodeAndReplaceLineBreaks(environment.OperatingSystem);
-            var osVersion = HtmlEncodeAndReplaceLineBreaks(environment.OperatingSystemVersion);
+            var osName = string.Empty;
+            var osVersion = string.Empty;
```
Comment (pranavkm) [inline src/Microsoft.AspNetCore.Hosting/Startup/StartupExceptionPage.cs]: Couldn't we just use the `OSDescription` field on `RuntimeInformation`? I think it includes the version

**Comment by pranavkm** [comment]
:watch:

File: src/Microsoft.AspNetCore.Hosting/Startup/StartupExceptionPage.cs (hunk 1 above)
Comment (davidfowl) [inline src/Microsoft.AspNetCore.Hosting/Startup/StartupExceptionPage.cs]: We should see how it looks on every os

Comment (pranavkm) [inline src/Microsoft.AspNetCore.Hosting/Startup/StartupExceptionPage.cs]: Windows: https://github.com/dotnet/corefx/blob/master/src/Common/src/Interop/Windows/NtDll/Interop.RtlGetVersion.cs#L21

Unix: https://github.com/dotnet/corefx/blob/d0dc5fc099946adc1035b34a8b1f6042eddb0c75/src/Native/System.Native/pal_runtimeinformation.cpp#L21

Seems better than what we have here.

Comment (BrennanConroy) [inline src/Microsoft.AspNetCore.Hosting/Startup/StartupExceptionPage.cs]: Changed to OSDescription and it shows:
```
<footer>
    .NET Core x64 v4.1.0.0&nbsp;&nbsp;&nbsp;|&nbsp;&nbsp;&nbsp;Microsoft.AspNetCore.Hosting version 1.0.0-rc2-20896&nbsp;&nbsp;&nbsp;|&nbsp;&nbsp;&nbsp;ubuntu 15.04&nbsp;&nbsp;&nbsp;|&nbsp;&nbsp;&nbsp;<a href="http://go.microsoft.com/fwlink/?LinkId=517394">Need help?</a>
</footer>
```
on Ubuntu
```
.NET Core X64 v4.1.0.0   |   Microsoft.AspNetCore.Hosting version 1.0.0-t002b27c25   |   Microsoft Windows 10.0.10586    |   Need help?
```
On Windows

**Comment by BrennanConroy** [comment]
🆙 📅

**Comment by pranavkm** [comment]
cc @Tratcher in case we're missing a scenario. Look :shipit: to me.

**Comment by Tratcher** [comment]
:+1:

**Pull request merged**